### PR TITLE
switch `flake8` to `ruff`

### LINKED ISSUE #868: switch to Ruff instead of `flake8`
### Description

Recently [Ruff](https://github.com/astral-sh/ruff) has become very popular among also mainstream projects for:
- much faster, they say 70x
- integrate several other tools under one roof, such as isort or bandit

### Use case

if you are fine with this change I am happy to send such PR for review... :rabbit:

### PR DESCRIPTION
**Pull request by Borda**
resolves #868

cc: @jmoralez

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[tool.ruff]
+target-version = "py38"
+line-length = 120
+# Enable Pyflakes `E` and `F` codes by default.
+select = [
+#    "E",
+#    "W", # see: https://pypi.org/project/pycodestyle
```
Comment (Borda) [inline pyproject.toml:7]: In the following PRs, I would suggest applying a few more basic checks

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[tool.ruff]
+target-version = "py38"
+line-length = 120
```
Comment (jmoralez) [inline pyproject.toml:3]: We use black's default of 88
```suggestion
line-length = 88
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "F", # see: https://pypi.org/project/pyflakes
+#    "I", #see: https://pypi.org/project/isort/
+#    "D", # see: https://pypi.org/project/pydocstyle
+#    "N", # see: https://pypi.org/project/pep8-naming
+#    "S", # see: https://pypi.org/project/flake8-bandit
+]
+
+
+[tool.mypy]
+ignore_missing_imports = "True"
```
Comment (jmoralez) [inline pyproject.toml:17]: Not sure if it makes a difference
```suggestion
ignore_missing_imports = true
```

**Review by jmoralez — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (Borda) [reply pyproject.toml:17]: That may not be valid yaml syntax

File: pyproject.toml (hunk 2 above)
Comment (Borda) [reply pyproject.toml:3]: Right, but as you check only for `F` and no `E` issues, long lines are skipped already :)
So this change has no effect and to be clear, it did not also have now...
just with enabling check for long lines, there are a few tens or hundreds issue :rabbit:

File: pyproject.toml (hunk 3 above)
Comment (jmoralez) [reply pyproject.toml:17]: But it's TOML. You can see it should be lowercase [here](https://mypy.readthedocs.io/en/stable/config_file.html#example-pyproject-toml).

**Review by jmoralez — APPROVED** [review]

**Pull request merged**